src/client/dfs/dfs.c

```diff
@@ -850,18 +850,25 @@ open_symlink(dfs_t *dfs, daos_handle_t th, dfs_obj_t *parent, int flags,
 		if (exists)
 			return EEXIST;
 
-		entry.value = (char *)value;
+		if (value == NULL || strnlen(value, PATH_MAX-1) > PATH_MAX-1)
```

Comment:
Good catch, although we should probably move away from using PATH_MAX and just define our out constant instead.  PATH_MAX is a poorly thought out define and doesn't do what most people think it does, and in fact it's meaningless to use it that way.

Reply:
we can define our own constant, but that would need to change other places.
i'll do that in a different patch.